src/main/java/org/jabref/model/openoffice/util/OOResult.java

```diff
@@ -0,0 +1,128 @@
+package org.jabref.model.openoffice.util;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/*
+ * error cannot be null
+ * result cannot be null
+ *
+ * Void is not allowed for R, use OOVoidResult instead.
+ *
+ * Out of `isPresent()` and `isError()` exactly one is true.
+ */
+public class OOResult<R, E> {
+    private final Optional<R> result;
+    private final Optional<E> error;
+
+    /**
+     * Exactly one of the arguments should be Optional.empty()
+     *
+     * @param result
+     * @param error
+     */
+    private OOResult(Optional<R> result, Optional<E> error) {
+        this.result = result;
+        this.error = error;
+    }
+
+    /**
+     * @param result Null is not allowed.
+     */
+    public static <R, E> OOResult<R, E> ok(R result) {
+        return new OOResult<>(Optional.of(result), Optional.empty());
+    }
+
+    /**
+     * @param error Null is not allowed.
+     */
+    public static <R, E> OOResult<R, E> error(E error) {
+        return new OOResult<>(Optional.empty(), Optional.of(error));
+    }
```

Comment:
When not using Optionals as method parameters, the following code should do it:

```suggestion
    private <R, E> OOResult<R, E> of(E error) {
        this.result = Optional.empty();
        this.error = Optional.of(error);
    }

    public static <R, E> OOResult<R, E> of(E error) {
        return new OOResult<>(error);
    }
```

Reply:
I assume the `of` in the green line 40 was meant to be `OOResult` (constructor)

Following that line of thought I got to

```
    private OOResult(Optional<R> result) {
        this.result = result;
        this.error = Optional.empty();
    }

    private OOResult(Optional<E> error) {
        this.result = Optional.empty();
        this.error = error;
    }
```

I think this would be ambiguous if the types E and R are related.
What if both are Integer? Did I misunderstand the suggestion?

Green line 45: `of` ? I assume this comes from copying, not intended.

Keeping as is, for now. Need clarification.